app/models/assignment.rb

```diff
@@ -957,12 +957,9 @@ def current_grader_data
       groups[[group_id, group_name, count]]
       groups[[group_id, group_name, count]] << ta unless ta.nil?
     end
-    # TODO: improve the group_sections calculation.
-    # In particular, this should be unified with Grouping#section.
     group_sections = {}
-    self.groupings.includes(:accepted_students).find_each do |g|
-      s = g.accepted_students.first
-      group_sections[g.id] = s&.section_id
+    self.groupings.each do |g|
```

Comment:
Use `.groupings.includes(:section)` to prevent an N+1 query (read about these here: https://guides.rubyonrails.org/active_record_querying.html#eager-loading-associations).